service/controller/drainer.go

```diff
@@ -31,41 +30,29 @@ func NewDrainer(config DrainerConfig) (*Drainer, error) {
 
 	var err error
 
-	var v1ResourceSet *controller.ResourceSet
+	var resourceSet []resource.Interface
 	{
-		c := v1.DrainerResourceSetConfig{
-			K8sClient: config.K8sClient,
-			Logger:    config.Logger,
-		}
-
-		v1ResourceSet, err = v1.NewDrainerResourceSet(c)
+		resourceSet, err = NewDrainerResourceSet(DrainerResourceSetConfig(config))
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 	}
 
-	var v2ResourceSet *controller.ResourceSet
+	var operatorkitController *controller.Controller
 	{
-		c := v2.DrainerResourceSetConfig{
-			K8sClient: config.K8sClient,
-			Logger:    config.Logger,
+		resourceSets := [][]resource.Interface{
```

Comment:
It seems like this is unnecessary. You could just pass in the slice of resources directly.